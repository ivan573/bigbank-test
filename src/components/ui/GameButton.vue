<template>
  <button
    class="game-button"
    :class="{'game-button_primary': primary}"
    v-on="$listeners"
    @click="handleButtonClick"
  >
    <span class="game-button__text">{{ text }}</span>
  </button>
</template>

<script>
export default {
  name: 'GameButton',
  props: {
    text: String,
    linkTo: String,
    primary: Boolean
  },
  methods: {
    handleButtonClick() {
      const path = this.linkTo
      path && path !== this.$route.path && this.$router.push({ path })
    }
  }
}
</script>

<style lang="scss" scoped>
  .game-button {
    padding: 8px;
    border: none;
    border-radius: 4px;
    background-color: black;
    cursor: pointer;
    display: block;

    &:hover {
      opacity: 0.5;
    }

    &__text {
      text-decoration: none;
      color: white;
    }

    &_primary {
      border: $border;
      background-color: white;
    }

    &_primary &__text {
      color: black;
    }
  }
</style>
