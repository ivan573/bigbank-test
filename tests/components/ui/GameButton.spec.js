import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import GameButton from '@/components/ui/GameButton.vue'

describe('GameButton.vue', () => {
  it('matches snapshot', () => {
    const text = 'text'
    const wrapper = shallowMount(GameButton, {
      propsData: { text }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('displays text got as prop', () => {
    const text = 'text'
    const wrapper = shallowMount(GameButton, {
      propsData: { text }
    })

    const textElement = wrapper.find('.game-button__text')
    expect(textElement.text()).toMatch(text)
  })

  it('has the primary class if gets the primary prop', () => {
    const wrapper = shallowMount(GameButton, {
      propsData: { primary: true }
    })

    expect(wrapper.classes('game-button_primary')).toBe(true)
  })

  it('does not have the primary class if does not get the primary prop', () => {
    const wrapper = shallowMount(GameButton)

    expect(wrapper.classes('game-button_primary')).toBe(false)
  })

  it('redirects to a path if a path is provided', () => {
    const localVue = createLocalVue()
    localVue.use(VueRouter)

    const router = new VueRouter()

    const linkTo = '/path'

    const wrapper = shallowMount(GameButton, {
      localVue,
      router,
      propsData: { linkTo }
    })

    wrapper.trigger('click')

    expect(wrapper.vm.$route.path).toBe(linkTo)
  })

  it('does not redirect if the path matches current path', () => {
    const $router = {
      push: jest.fn()
    }

    const linkTo = '/path'

    const $route = {
      path: linkTo
    }

    const wrapper = shallowMount(GameButton, {
      mocks: {
        $route,
        $router
      },
      propsData: { linkTo }
    })

    wrapper.trigger('click')

    expect(wrapper.vm.$router.push).toHaveBeenCalledTimes(0)
  })

  it('emits click event on click', () => {
    const handleClick = jest.fn()

    const wrapper = shallowMount(GameButton, {
      listeners: {
        click: handleClick
      }
    })

    wrapper.trigger('click')

    expect(handleClick).toHaveBeenCalled()
  })
})
