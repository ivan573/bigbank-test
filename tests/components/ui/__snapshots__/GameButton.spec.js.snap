// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameButton.vue matches snapshot 1`] = `
<button
  class="game-button"
>
  <span
    class="game-button__text"
  >
    text
  </span>
</button>
`;
